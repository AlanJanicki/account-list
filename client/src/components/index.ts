export { Card } from './Card/Card';
export { Loader } from './Loader/Loader';
export { Spinner } from './Spinner/Spinner';
export * from './Table/Table';
