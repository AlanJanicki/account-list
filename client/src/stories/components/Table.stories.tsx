import { Table } from '../../components';
import { getCellsWithMockedData } from '../../mocks/accountList';

export default {
  component: Table,
  tags: ['autodocs'],
  title: 'Table'
};

export const Default = {
  args: {
    cells: getCellsWithMockedData()
  }
};
