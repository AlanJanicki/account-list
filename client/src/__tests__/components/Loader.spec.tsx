import React from 'react';

import { Loader } from '../../components';
import { renderWithSnapshot } from '../common';

describe('<Loader/>', () => {
  it('matches snapshot - loading', () => renderWithSnapshot(<Loader isLoading />));
  it('matches snapshot - isError', () =>
    renderWithSnapshot(
      <Loader
        isError
        button={{ onClick: jest.fn(), text: 'Try again' }}
        error='Something went wrong'
        isLoading={false}
      />
    ));
});
