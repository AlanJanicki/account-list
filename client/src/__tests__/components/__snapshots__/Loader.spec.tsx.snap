// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Loader/> matches snapshot - isError 1`] = `
<DocumentFragment>
  .c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  text-align: center;
}

.c1 {
  color: #dd3546;
}

.c2 {
  margin: 2rem auto 0 auto;
  padding: 0.5rem;
  width: 50%;
  background-color: #0f7bff;
  border: 1px solid #dfdfdf;
  border-radius: .25rem;
  border: 1px solid transparent;
  color: #fff;
  font-weight: 400;
}

@media (min-width:992px) {
  .c1 {
    font-size: 1.2rem;
  }
}

@media (min-width:992px) {
  .c2 {
    font-size: 1.2rem;
  }
}

<div
    class="c0"
  >
    <div>
      <span
        class="c1"
      >
        Something went wrong
      </span>
      <button
        class="c2"
      >
        Try again
      </button>
    </div>
  </div>
</DocumentFragment>
`;

exports[`<Loader/> matches snapshot - loading 1`] = `
<DocumentFragment>
  .c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  text-align: center;
}

.c1 {
  width: 50px;
  display: block;
  aspect-ratio: 1;
  border: 3px solid #0f7bff;
  border-radius: 50%;
  border-top-color: #fff;
  -webkit-animation: spin 1s ease-in-out infinite;
  animation: spin 1s ease-in-out infinite;
}

<div
    class="c0"
  >
    <p
      role="status"
    >
      <span
        class="c1"
        width="50"
      />
    </p>
  </div>
</DocumentFragment>
`;
